### ISSUE TITLE
Add redirects for old documentation URLs that are now 404s

### ISSUE THREAD

**Issue by davidxia** [kubernetes/website#21429]
**This is a Feature Request**

**What would you like to be added**
URLs like https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.15 used to work. Now they're 404s. It would be great to have redirects for these to the new URLs at https://v1-15.docs.kubernetes.io/docs/reference/generated/kubernetes-api/v1.15/ if it's not too much work.

**Why is this needed**
Backwards compatibility and convenience.

**Comments**
I'm not sure how many versions have links like that were recently removed. Would be great to have a reasonable amount of them covered by redirects.

**Comment by sftim** [comment]
/help
/priority backlog

I'd go back only as far back as the version selector in the documentation website does. Issue #12303 tracks problems with links to even older websites. It'd be nice to serve those as _410 Gone_ responses, but that's not feasible with the current hosting setup.

**Comment by KilimAnnejaro** [comment]
@sftim @davidxia is there a list of links that lead to 404s, or is part of this story tracking down all of those links?

**Comment by davidxia** [comment]
@KilimAnnejaro sorry I don't have a full list. I only noticed the links like the one in my description. I think creating the list is part of this issue.

**Comment by kbhawkey** [comment]
@KilimAnnejaro ,
The site lists five versions. The current version of the docs is 1.18.
You could add redirects for the previous four versions (1.17, 1.16, 1.15, 1.14)

```
# example _redirects
/docs/reference/generated/kubernetes-api/v1.17/    https://v1-17.docs.kubernetes.io/docs/reference/generated/kubernetes-api/v1.17/  301

```

**Comment by KilimAnnejaro** [comment]
I generated a list of pages with links leading to 404s using https://www.brokenlinkcheck.com/. I've attached a PDF of this list:

https://drive.google.com/file/d/1-CEM0UHYjv3qdwRKcINgQg8OK1tqH7Kh/view?usp=sharing

Of course, not all of these links are directly related to the thrust of this issue; however, the list seemed useful so I'm attaching it here.

**Comment by sftim** [comment]
Partly duplicated by issue #21583

**Comment by sftim** [comment]
If anyone's picking this up, I recommend going with the approach that @kbhawkey outlines in https://github.com/kubernetes/website/issues/21429#issuecomment-639818071

**Comment by kbhawkey** [comment]
@KilimAnnejaro , Do you need a hand with this issue? I am happy to help out.

**Comment by kbhawkey** [comment]
I've gone ahead and created a PR for this issue. Will close my PR if someone else
has fixed this issue. I feel obligated to get this issue resolved as I cleaned up the API references and contributed to the redirect issues.
See PR #21826.

**Comment by KilimAnnejaro** [comment]
@kbhawkey feel free to remove me as assigned to this issue. I was going to work on this on an evening/over the weekend (I'm only part-time on k8s), but if you need the fix sooner, I'm probably not the person for this.

**Comment by kbhawkey** [comment]
Thanks @KilimAnnejaro

**Comment by tengqm** [comment]
fixed by #21826 
/close
